NetKVM/wlh/ParaNdis6-Oid.cpp

```diff
@@ -703,8 +703,8 @@ void ParaNdis6_FillOffloadConfiguration(PARANDIS_ADAPTER *pContext)
     NDIS_OFFLOAD *po = &pContext->ReportedOffloadConfiguration;
     FillOffloadStructure(po, pContext->Offload.flags);
 #if PARANDIS_SUPPORT_RSC
-    po->Rsc.IPv4.Enabled = (pContext->RSC.bIPv4SupportedSW && pContext->RSC.bIPv4SupportedHW);
```

Comment:
HI Yan,

If the backend doesn't support RSC, protocol stack/overlay driver won't turn it ON. The problem is that the system might request to turn if off if the netkvm/backend set it as ON. One of the reason is that network filter driver doesn't support RSC and then need to turn it off by sending a OID_TCP_OFFLOAD_PARAMETERS request. In current code, we return NDIS_SUCCESS while still indicating RSC as enabled. The several filter driver registration/startup will just hang in there forever.

You can observe it by installing docker on a VM with RSC enabled. By default, docker will create NAT vswitch. winnat.sys will attach a filter driver which in turn will try to turn off RSC. (docker won't start).